pkg/curl/schemes.go

```diff
@@ -219,25 +203,38 @@ func (t *TFTPClient) Fetch(_ context.Context, u *url.URL) (io.ReaderAt, error) {
 	return uio.NewCachingReader(r), nil
 }
 
-// RetryFilter implements FileSchemeRetryFilter.
-func (t *TFTPClient) RetryFilter(u *url.URL, err error) bool {
-	// The tftp does not export the necessary structs to get the
-	// code out of the error message cleanly.
+// RetryTFTP retries downloads if the error does not contain FILE_NOT_FOUND.
+//
+// pack.ag/tftp does not export the necessary structs to get the
+// code out of the error message cleanly, but it does embed FILE_NOT_FOUND in
+// the error string.
+func RetryTFTP(u *url.URL, err error) bool {
 	return !strings.Contains(err.Error(), "FILE_NOT_FOUND")
 }
 
+// DoRetry returns true if the Fetch request for the URL should be
+// retried. err is the error that Fetch previously returned.
+//
+// DoRetry lets a FileScheme filter for errors returned by Fetch
+// which are worth retrying. If this interface is not implemented, the
+// default for SchemeWithRetries is to always retry. DoRetry
+// returns true to indicate a request should be retried.
+type DoRetry func(u *url.URL, err error) bool
+
 // SchemeWithRetries wraps a FileScheme and automatically retries (with
 // backoff) when Fetch returns a non-nil err.
 type SchemeWithRetries struct {
 	Scheme  FileScheme
+	DoRetry DoRetry
```

Comment:
Mention whether the default is the retry if this is not set.

Reply:
done.